File: httpsnippet/src/test/resources/output/clojure/clj_http/multipart-file.clj
(require '[clj-http.client :as client])

(client/post "http://mockbin.com/har" {headers {content-type "multipart/form-data" 
                                         } 
                                      :multipart [{name "foo" 
                                           :content (clojure.java.io/file "hello.txt") 
                                           } ] 
                                      })

File: httpsnippet/src/test/resources/output/javascript/jquery/multipart-file.js
var form = new FormData();
form.append("foo", "hello.txt");

var settings = {"async":true,"headers":{"content-type":"multipart/form-data"},"processData":false,"method":"POST","data":form,"crossDomain":true,"mimeType":"multipart/form-data","contentType":false,"url":"http://mockbin.com/har"}

$.ajax(settings).done(function (response) {
  console.log(response);
});